quay/quay-operator
# Added support for environment variables and specifying a password for redis

Added functionality to define environment variables in each of the primary components. 

To Test:

1. Update the CRD with the contents in the `deploy/crds/redhatcop_v1alpha1_quayecosystem_crd.yaml` file
2. Deploy the operator
   1. Use the following Operator image which contains the changes introduced in this PR: `quay.io/ablock/quay-operator:c39de4def081c8c0c857578ed2edcb198f1a04d3`
3. Create a new Redis Password Secret

```
oc create secret generic redis-credentials --from-literal=password=<password>
```

4. Create an ImagePullSecret to retrieve protected images
5. Create a `QuayEcosystem` Custom Resource

```
apiVersion: redhatcop.redhat.io/v1alpha1
kind: QuayEcosystem
metadata:
  name: example-quayecosystem
spec:
  quay:
    imagePullSecretName: <pull_secret>
    envVars:
      - name: FOO
         value: bar
  redis:
    credentialsSecretName: <secret_name>
    imagePullSecretName: <pull_secret>
```

Once deployed verify redis is making use of a password as well as the environment variable is defined within the quay container
